data/build_movielens.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2020 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script for building the movielens tags/sequences from unformatted dataset"""
+import os
+import json
+import functools
+import glob
+import collections
+import numpy as np
+from absl import app, logging, flags
+import pandas as pd
+import tqdm
+from sklearn.model_selection import train_test_split
+
+
+flags.DEFINE_string("movielens_dir", "./data/movielens", \
+  "path to the movielens folder")
+flags.DEFINE_enum("dataset", "both", ["tags", "sequences", "both"], \
+  "which dataset to generate from movielens: tags, movie sequences, or both")
+flags.DEFINE_string("output_dir", "gs://e2e_central/data", "path to write TSVs")
+flags.DEFINE_enum("task", "all", ["ml_tags", "ml_sequences", "all"], \
+  "task to build datasets for: ml_tags, ml_sequences, or all")
+flags.DEFINE_bool("mask", False, \
+  "boolean create masked version of ml_tags dataset")
+flags.DEFINE_integer("sample_rate", 3, \
+  "rate for sampling multiple masked examples from one ml_tags example")
+FLAGS = flags.FLAGS
+
+def main(_):
+  """Builds the movielens datasets and saves tsvs in output_dir"""
+  # Define filepaths
+  paths = {
+      "sequences": sorted(glob.glob(os.path.join(FLAGS.movielens_dir,
+                                                 "sequences", "*.csv*"))),
+      "movies": glob.glob(os.path.join(FLAGS.movielens_dir,
+                                       "ml-25m/movies.csv"))[0],
+      "tags": glob.glob(os.path.join(FLAGS.movielens_dir,
+                                     "ml-25m/tags.csv"))[0],
+      "genome_tags": glob.glob(os.path.join(FLAGS.movielens_dir,
+                                            "ml-25m/genome-tags.csv"))[0],
+      "genome_scores": glob.glob(os.path.join(FLAGS.movielens_dir,
+                                              "ml-25m/genome-scores.csv"))[0]
+  }
+
+  # Load the movie and tag id files
+  genome_tags = pd.read_csv(paths["genome_tags"])
+  tag_decoder = dict(zip(genome_tags.tagId, genome_tags.tag))
+  movies = pd.read_csv(paths["movies"])
+  movies["title"] = movies["title"].apply(flip_titles)
+  movie_decoder = dict(zip(movies.movieId, movies.title))
+  genre_decoder = dict(zip(movies.title, movies.genres))
+
+  if FLAGS.task in ["ml_sequences", "all"]:
```

Comment:
Suggest to wrap in a function and explicitly call out the arguments, as this if is growing a bit big. Same for the other if below.